src/cloudAdapter/cluster/clusterAdapter.ts

```diff
@@ -0,0 +1,147 @@
+            // docker tag SOURCE_IMAGE[:TAG] harbor.dev.cluster.genez.io/9b59dd97-6c04-40f4-bdb2-1bba31da4e31/REPOSITORY[:TAG]
+            const dockerTag = spawnSync(
+                "docker",
+                [
+                    "tag",
+                    `${projectConfiguration.name}-${element.name}:latest`.toLowerCase(),
+                    `${registryCreds.repository}/${registryCreds.username}/${element.name}`.toLowerCase(),
+                ],
+                { stdio: "inherit" },
+            );
+            if (dockerTag.status !== 0) {
+                throw new Error(`Error during docker tag. Exit code: ${dockerTag.status}`);
+            }
+            // push image
+            // docker push harbor.dev.cluster.genez.io/9b59dd97-6c04-40f4-bdb2-1bba31da4e31/REPOSITORY[:TAG]
+            const dockerPush = spawnSync(
+                "docker",
+                [
+                    "push",
+                    `${registryCreds.repository}/${registryCreds.username}/${element.name}`.toLowerCase(),
+                ],
+                { stdio: "inherit" },
+            );
+            if (dockerPush.status !== 0) {
+                throw new Error(`Error during docker push. Exit code: ${dockerPush.status}`);
+            }
+            debugLogger.debug(`Done uploading the container image for ${element.name}.`);
+        });
+
+        await Promise.all(promisesDeploy);
+
+        const response = await deployRequest(projectConfiguration, stage);
+        const classesInfo = response.classes.map((c) => ({
+            className: c.name,
+            methods: c.methods.map((m) => ({
+                name: m.name,
+                type: m.type,
+                cronString: m.cronString,
+                functionUrl: getFunctionUrl(c.cloudUrl, m.type, c.name, m.name),
+            })),
+            functionUrl: c.cloudUrl,
+            projectId: response.projectId,
+        }));
+
+        console.log(classesInfo);
+        return {
+            projectEnvId: "",
+            classes: classesInfo,
+        };
+    }
+
+    deployFrontend(
+        projectName: string,
+        projectRegion: string,
+        frontend: YamlFrontend,
+        stage: string,
+    ): Promise<string> {
+        throw new Error("Method not implemented.");
+    }
+}
+
+function getFunctionUrl(
+    baseUrl: string,
+    methodType: string,
+    className: string,
+    methodName: string,
+): string {
+    if (methodType === "http") {
+        // trim the last slash of baseUrl if it exists
+        if (baseUrl.endsWith("/")) {
+            baseUrl = baseUrl.slice(0, -1);
+        }
+        return `${baseUrl}/${className}/${methodName}`;
+    } else {
+        return `${baseUrl}/${className}`;
+    }
+}
+
+function getDockerPasswordFromGenezioRc() {
+    const homeDir = os.homedir();
+    const genezioRcPath = path.join(homeDir, ".geneziorc");
+    const genezioRc = fs.readFileSync(genezioRcPath, "utf8");
+    return `HRBR${genezioRc.substring(0, 64)}`;
+}
```

Comment:
I think this very similar to `getAuthToken`[1], maybe you can reuse it to avoid duplicated code?

[1] https://github.com/Genez-io/genezio/blob/66eb9106d2e92c3c09632d1c20cb95a0a82aae6b/src/utils/accounts.ts#L9C1-L9C1